airflow/providers/google/cloud/transfers/gdrive_to_gcs.py

```diff
@@ -0,0 +1,134 @@
+
+
+class GoogleDriveToGCSOperator(BaseOperator):
+    """
+    Writes a Google Drive file into Google Cloud Storage.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:GoogleDriveToGCSOperator` FIXME
+
+    :param destination_bucket: The destination Google cloud storage bucket where the
+        report should be written to. (templated)
+    :param destination_bucket: str
+    :param destination_object: The Google Cloud Storage object name for the object created by the operator.
+        For example: ``path/to/my/file/file.txt``.
+    :type destination_object: str
+    :param folder_id: The folder id of the folder in whhich the Google Drive file resides
+    :type folder_id: str
+    :param file_name: The name of the file residing in Google Drive
+    :type file_name: str
+    :param gcp_conn_id: The GCP connection ID to use when fetching connection info.
+    :type gcp_conn_id: str
+    :param delegate_to: The account to impersonate using domain-wide delegation of authority,
+        if any. For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :type delegate_to: str
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account (templated).
+    :type impersonation_chain: Union[str, Sequence[str]]
+    """
+
+    template_fields = [
+        "destination_bucket",
+        "destination_object",
+        "folder_id",
+        "file_name",
+        "impersonation_chain",
+    ]
+
+    @apply_defaults
+    def __init__(
+        self,
+        *,
+        destination_bucket: str,
+        destination_object: str,
+        folder_id: Optional[str] = None,
+        file_name: Optional[str] = None,
+        gcp_conn_id: str = "google_cloud_default",
+        delegate_to: Optional[str] = None,
+        impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
+        **kwargs,
+    ) -> None:
+        if folder_id and not file_name:
+            raise AirflowException("Both folder_id and file_name must be set. Received only folder_id")
+        if not folder_id and file_name:
+            raise AirflowException("Both folder_id and file_name must be set. Received only file_name")
+        if not folder_id and not file_name:
+            raise AirflowException("Both folder_id and file_name must be set. Received neither.")
```

Comment:
Why do not make them required and avoid this checks?

Reply:
Good point